### ISSUE TITLE
H39 Using caption elements to associate data table captions with data tables needs a rewritten test

### ISSUE THREAD

**Issue by mbgower** [w3c/wcag#2136]
H39 is considered sufficient to meet an aspect of 1.3.1; however, the context needs to be better expressed, and the test rewritten to reflect the fact that a caption for a table is not needed. 

[1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships) states:
> Information, structure, and relationships conveyed through presentation can be programmatically determined or are available in text.

If text exists that seems to provide a name or title for a table, then 1.3.1 requires that there be programmatic indication and proper association for the caption.. That's where H39 comes from. 

But if there is no visible text name/title for the table, then there is nothing conveyed through presentation. Therefore 1.3.1 doesn't apply.

Rather than a 3-step test procedure, the test should be written as a conditional:

Existing
>For each data table:
Check that the table has content that is presented as a table caption.
Check that the table includes a caption element.
Check that the content of the caption element identifies the table.

Proposed
For each data table that has content that is presented as a table title or caption:
1. Check that the table includes a caption element.
2. Check that the content of the caption element identifies the table.

**Comment by mbgower** [comment]
I created a pull request that makes these changes. I also adjusted the wording of the original third bullet in the procedure, because H39 does not cover the quality of the text used to title a table; only that the text be coded as a caption

**Comment by awkawk** [comment]
@mbgower I don't think that there is anything wrong with H39 as written. IF you want to apply the technique, then all three checks must pass. If you don't want to apply the technique (possibly because the table is a layout table, possibly because the table doesn't need a caption) that is ok too, because those won't meet the technique (the same is true for the original or modified wording).

**Comment by awkawk** [comment]
Also, how did you change the final bullet? It looks the same?

**Comment by patrickhlauke** [comment]
to me #2137 makes sense, as it doesn't substantially change anything, but just adds a reminder of what the scope of the technique itself is (i.e. `The objective of this technique is to programmatically associate captions for data tables where captions are provided in the presentation.`) and gives it as a prerequisite to running the tests, rather than seemingly implying that all tables need that (a reader just skimming this technique may indeed get that impression)

**Comment by mbgower** [comment]
I agree it is possible to read it that way. But I also have someone arguing that this procedure means you need to have a caption, period.

I don't want that to be inferred, and I also think the wording of the third bullet over-reached what this technique is for.

**Comment by mbgower** [comment]
> Also, how did you change the final bullet? It looks the same?

New wording (with additional commit) is:
> Check that the text that titles or describes the table is included in the <code class="el">caption</code> element.

**Comment by awkawk** [comment]
> But I also have someone arguing that this procedure means you need to have a caption, period.

They may still argue that. :)

If making a change, I think that we might want to stick to "title" as that is what the HTML spec uses for caption element content.

```
For each data table that has content that is presented as a title:
Check that the table includes a caption element
Check that the text that titles the table is included in the caption element.
```

**Comment by mbgower** [comment]
> I think that we might want to stick to "title"

Either is fine with me.

**Comment by mbgower** [comment]
@awkawk Are there any other tags I should add to this?

**Comment by awkawk** [comment]
@mbgower not sure what "this" means here ...

**Comment by mbgower** [comment]
any other tags to add to this issue? it could be 'wcag 2.2' but it could be for 2.x (for which there is no tag)

**Comment by awkawk** [comment]
Ah, you mean GitHub labels, got it. Two years ago the labels you have would have been fine. Chairs may have others to route now, not sure.

**Comment by alastc** [comment]
The WCAG 2.0/1/2 labels are used for which version the SC was introduced in, so in this case I've added WCAG 2.0. Helps to work out how many issues are raised or open against which SCs.

**Comment by patrickhlauke** [comment]
> The WCAG 2.0/1/2 labels are used for which version the SC was introduced in, so in this case I've added WCAG 2.0. Helps to work out how many issues are raised or open against which SCs.

hah, i clearly misunderstood these labels until now, as i thought it was more about which versions are affected, and/or in which version it's likely to get some kind of change made (keeping in mind that older WCAG versions are cast in amber for eternity)

**Comment by mgifford** [comment]
I just ran into this issue IRL. Do we need a PR for this to move it ahead?

**Comment by bruce-usab** [comment]
closed by #2137
